common/app/routes/Challenges/redux/index.js

```diff
@@ -220,7 +220,7 @@ export const challengeMetaSelector = createSelector(
   // use closure to get around circular deps
   (...args) => challengeSelector(...args),
   challenge => {
-    if (!challenge.id) {
+    if (!challenge || !challenge.id) {
       const viewType = 'invalid';
```

Comment:
If you are returning empty object as from above snippet. Is not !challenge condition will never evaluate to true?

Reply:
That's true. I tried removing the "!challenge" condition and it still works fine. Thanks for pointing that out. Please request a change and I'll update my PR

Reply:
You welcome. Sure.

Reply:
Your welcome.

Reply:
Need to remove unnecessarily condition.